tfutils/base.py

```diff
@@ -708,6 +711,19 @@ def _save_thread(self, save_filters_permanent, save_filters_tmp, save_rec, step,
                     outrec = putfs.put(_fp, filename=saved_path, **save_rec)
             log.info('... done putting filters into database.')
 
+            if not save_filters_permanent:
+                recent_gridfs_files = self.collfs_recent._GridFS__files
+                recent_query_result = recent_gridfs_files.find({'saved_filters': True}, sort=[('uploadDate', 1)])
+                num_cached_filters = recent_query_result.count()
+                cache_max_num = self.cache_max_num
+                if num_cached_filters > cache_max_num:
+                    log.info('Cleaning up cached filters')
+                    fsbucket = gridfs.GridFSBucket(recent_gridfs_files._Collection__database, bucket_name=recent_gridfs_files.name.split('.')[0])
+
+                    for del_indx in xrange(0, num_cached_filters - cache_max_num):
+                        #log.info(recent_query_result[del_indx]['uploadDate'])
+                        fsbucket.delete(recent_query_result[del_indx]['_id'])
```

Comment:
@chengxuz

(1) does this delete the stuff in the filesystem while still retaining the record?  i think thats what we want, right?  we still want the record of performance even if we remove the filters

(2) in which case we'd then have to set "saved_filters" to False.

Reply:
Yes, it will. I am deleting from __RECENT gfs files (essentially another database). And they are independent from main gfs files. The model will be put into main gfs files only when the batch number is a multiple of save_filter_freq. And all the records are saved in main gfs files. So my deletion will not influence the records.